relayer/src/services/arbitrum.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  L2TransactionReceipt,
+  L2ToL1MessageStatus,
+  L1ToL2MessageGasEstimator,
+  L1TransactionReceipt,
+  L1ToL2MessageStatus,
+} from "@arbitrum/sdk";
+import type { Signer } from "@ethersproject/abstract-signer";
+import { BaseRelayerService } from "./types";
+import ArbitrumL1MessengerABI from "../assets/abi/ArbitrumL1Messenger.json";
+import { BigNumber, Contract } from "ethers";
+
+const L1_CONTRACTS: Record<number, { ARBITRUM_L1_MESSENGER: string }> = {
+  42161: {
+    ARBITRUM_L1_MESSENGER: "0x7230a5D821Bd00453344b750dff43FAC754A519E",
+  },
+  421613: {
+    ARBITRUM_L1_MESSENGER: "0xCaf560790596E05abcDb6dA9A0ABfaf216bcAF8b",
+  },
+};
+
+export class ArbitrumRelayerService extends BaseRelayerService {
+  static readonly MAX_MESSAGE_LENGTH_BYTES = 1_000_000;
+
+  async prepare(): Promise<boolean> {
+    console.log("Preparing Arbitrum Messenger for forwarding the L1 message...");
+
+    const arbitrumL1Messenger = new Contract(
+      L1_CONTRACTS[await this.getL2ChainId()].ARBITRUM_L1_MESSENGER,
+      ArbitrumL1MessengerABI,
+      this.l1Wallet,
+    );
+
+    const currentDeposit: BigNumber = await arbitrumL1Messenger.deposits(this.l1Wallet.address);
+    const l1ToL2MessageGasEstimate = new L1ToL2MessageGasEstimator(this.l2RpcProvider);
+
+    const submissionFee = await l1ToL2MessageGasEstimate.estimateSubmissionFee(
+      this.l1RpcProvider,
+      BigNumber.from(0),
+      BigNumber.from(ArbitrumRelayerService.MAX_MESSAGE_LENGTH_BYTES),
+    );
+
+    const deltaSubmissionFee = submissionFee.mul(110).div(100); // We add 10% on top to ensure that the creation of a retryable ticket doesn't fail
+
+    const amountToDeposit = deltaSubmissionFee.sub(currentDeposit);
+
+    if (amountToDeposit.gt(0)) {
+      const estimatedGasLimit = await arbitrumL1Messenger.estimateGas.deposit({
+        value: amountToDeposit,
+      });
+
+      const transaction = await arbitrumL1Messenger.deposit({
+        gasLimit: estimatedGasLimit,
+        value: amountToDeposit,
+      });
+
+      await transaction.wait();
+
+      const receipt = await arbitrumL1Messenger.provider.waitForTransaction(transaction.hash, 1);
+
+      console.log(
+        `Successfully deposited funds on ArbitrumL1Messenger: ${receipt.transactionHash}`,
+      );
+    } else {
+      console.log("ArbitrumL1Messenger has enough funds to proceed!");
+    }
+
+    return true;
+  }
+
+  async relayTxToL1(l2TransactionHash: string): Promise<string | undefined> {
+    console.log("Arbitrum outbox execution");
+
+    /**
+   / * We start with a txn hash; we assume this is transaction that triggered an L2 to L1 Message on L2 (i.e., ArbSys.sendTxToL1)
+  */
+
+    if (!l2TransactionHash.startsWith("0x") || l2TransactionHash.trim().length != 66)
```

Comment:
This is only a small check but the same for all RelayerServices, so could probably be generalized in a way to not have code duplication.

Reply:
I moved the argument validations to its own function located in the `relayer-program.ts`. Thanks for catching this ✅